### ISSUE TITLE
`int_from_float` errors when saving a table of contents

### ISSUE THREAD

**Issue by kevinhu** [VikParuchuri/marker#417]
I'm saving the list of `TocItems` as part of a response, but I'm running into validation errors reporting that the polygon coordinates are floats rather than ints. It seems that `TocItem`'s polygon field expects a `list[list[int]]`, but it is being passed floats in its constructor.

https://github.com/VikParuchuri/marker/blob/8992482cadb91d4f8be46fea47cad844e1b62bcb/marker/schema/document.py#L18-L22

https://github.com/VikParuchuri/marker/blob/8992482cadb91d4f8be46fea47cad844e1b62bcb/marker/processors/document_toc.py#L16-L21

```
ValidationError: 62 validation errors for ProcessFileResponse
table_of_contents.0.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=287.2265625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.0.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=2.060546875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=554.4140625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.1.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=2.060546875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=554.4140625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=58.107421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=287.2265625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.0.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=58.107421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.1.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=224.8828125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.1.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=224.8828125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=52.32421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.0.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=418.703125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=494.296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.1.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=418.703125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=494.296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=471.453125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=52.32421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.2.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=471.453125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.3.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=137.490234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.3.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=275.2890625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.3.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=137.490234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.3.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=275.2890625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=382.96875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.0.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=219.2421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=498.75, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.1.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=219.2421875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=498.75, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=258.8046875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=382.96875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.4.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=258.8046875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.5.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=222.65625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.5.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=222.65625, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.5.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=306.609375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.5.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=306.609375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.6.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=220.4296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.6.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=220.4296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.7.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=347.34375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.7.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=253.859375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.7.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=347.34375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.7.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=253.859375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.0.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=178.03125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=496.5234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.1.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=178.03125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=496.5234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=250.5625, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.8.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=250.5625, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.9.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=64.5703125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.9.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=354.0234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.9.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=354.0234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.9.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=64.5703125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.0.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=327.3046875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.0.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=184.625, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=496.5234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.1.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=184.625, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=496.5234375, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=260.453125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.3.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=327.3046875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.10.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=260.453125, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.11.polygon.1.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=494.296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.11.polygon.2.0
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=494.296875, 
input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.11.polygon.2.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=263.75, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
table_of_contents.11.polygon.3.1
  Input should be a valid integer, got a number with a fractional 
part [type=int_from_float, input_value=263.75, input_type=float]
    For further information visit 
https://errors.pydantic.dev/2.9/v/int_from_float
```

**Comment by VikParuchuri** [comment]
Thanks, will fix this shortly

**Comment by iammosespaulr** [comment]
Thank you for reporting, this has been fixed by #422!
